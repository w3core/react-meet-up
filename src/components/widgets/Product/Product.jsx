import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';

const displayName = 'Product';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleRemove: PropTypes.func.isRequired,
};

function Product({ name, handleRemove }) {
  return (
    name && (
      <div data-name={name}>
        {name}
        {' - '}
        <Button onClick={handleRemove} value="removeProduct">
          Remove
        </Button>
      </div>
    )
  );
}

Product.displayName = displayName;
Product.propTypes = propTypes;

export default Product;
