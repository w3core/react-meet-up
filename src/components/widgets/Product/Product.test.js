import React from 'react';
import { shallow } from 'enzyme';
import Product from './Product';

describe('Given the Product component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<Product name="Product 1" handleRemove={jest.fn()} />);

    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
