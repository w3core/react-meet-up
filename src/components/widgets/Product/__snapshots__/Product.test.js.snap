// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Product component when the component is rendered should render the component with provided props 1`] = `
<div>
  Product 1
   - 
  <WithStyle(Button)
    onClick={[MockFunction]}
    value="removeProduct"
  >
    Remove
  </WithStyle(Button)>
</div>
`;
